src/JointTrajectoryControllerBase.cpp

```diff
@@ -241,13 +243,16 @@ void JointTrajectoryControllerBase::updateStep(const ros::Time &time,
     trajectory->evaluateDerivative(timeFromStart, 1, mDesiredVelocity);
     trajectory->evaluateDerivative(timeFromStart, 2, mDesiredAcceleration);
 
-    // TODO: Check path constraints.
+    std::string stopReason;
 
-    // Check goal constraints.
+    // Check trajectory and goal constraints.
+    bool trajConstraintsSatisfied = true;
     bool goalConstraintsSatisfied = true;
     for (const auto &dof : mControlledSkeleton->getDofs()) {
+      auto trajIt = mTrajectoryConstraints.find(dof->getName());
       auto goalIt = mGoalConstraints.find(dof->getName());
-      if (goalIt != mGoalConstraints.end()) {
+      if (trajIt != mTrajectoryConstraints.end() ||
```

Comment:
You need to handle this logic in two separate if statements rather than a combined if statement.

In this if body, either `trajIt` or `goalIt` could be null (just not both). However, you don't check this later on when you call `trajIt->second`, which can cause a segfault. Therefore, this doesn't handle a lack of trajectory constraints gracefully.

Reply:
Oops, thanks for catching this! Looks like I introduced this when I tried to refactor the logic out of `shouldStopExecution`.